### PR TITLE
Include pytorch_export_contrib_ops in inference builds

### PR DESCRIPTION
**Pull request by garymm**
Rename / move it from tools/python/register_custom_ops_pytorch_exporter
to onnxruntime/python/tools/pytorch_export_contrib_ops.

Rationale for inclusion in inference builds:
This code is potentially useful for anyone using ORT, not just training.

Rationale for new name:
"Contrib op" is the nomenclature used within ORT to refer to the set of
ops that are not in the standard op set but are included by default with
ORT. This is more specific than "custom op", which is what the PyTorch
exporter uses to refer to any non-standard op.

Step 1 of addressing #8818. After this is merged I will update the docs.

### CODE REVIEW

**Comment by wenbingl** [comment]
> Rename / move it from tools/python/register_custom_ops_pytorch_exporter
> to onnxruntime/python/tools/pytorch_export_contrib_ops.
> 
> Rationale for inclusion in inference builds:
> This code is potentially useful for anyone using ORT, not just training.
> 
> Rationale for new name:
> "Contrib op" is the nomenclature used within ORT to refer to the set of
> ops that are not in the standard op set but are included by default with
> ORT. This is more specific than "custom op", which is what the PyTorch
> exporter uses to refer to any non-standard op.
> 
> Step 1 of addressing #8818. After this is merged I will update the docs.

IMO, ContribOp was used to mainly solve the performance issue, like fusing some operators into one contrib-op for better performance. This name may not be very accurate now, due to its history.

Runtime team has spent some efforts to create CustomOp API to enable non-standard ONNX operator work, which remove the op development dependency on a lot of ONNXRuntime internal stuff, and this kind of dependency creates some extra maintenance cost.

In this case, These PyTorch ops may leverage CustomOp API as well, and should not be considered as 'ContribOp'.
@pranavsharma for more comments.

File: onnxruntime/python/tools/pytorch_export_contrib_ops.py [hunk 1]
```diff
@@ -1,21 +1,38 @@
...
-_onnx_opset_version = 1
+__OPSET_VERSION = 1
+__registered_ops: typing.AbstractSet[str] = set()
 
 
-def register_custom_op():
-    """
-    This function registers symbolic functions for
-    custom ops that are implemented as part of ONNX Runtime
+def __reg(symbolic_fn: typing.Callable):
```
Comment (thiagocrepaldi) [inline onnxruntime/python/tools/pytorch_export_contrib_ops.py:24]: Name is a little vague. We usually use single `_` to refer to private stuff in ORT. Double `_` is usually reserved for python internal names

**Review by thiagocrepaldi — COMMENTED** [review]
LGTM.

**Comment by garymm** [comment]
@wenbingl what is the name for the non-standard ops that are included by default in ORT official releases?
Those are the ops that I think should be registered by this code, and I'll use that name.

@thiagofc I believe you didn't actually approve in the eyes of GitHub. If you meant to approve, please do so.

**Comment by pranavsharma** [comment]
> @wenbingl what is the name for the non-standard ops that are included by default in ORT official releases?
> Those are the ops that I think should be registered by this code, and I'll use that name.
> 
> @thiagofc I believe you didn't actually approve in the eyes of GitHub. If you meant to approve, please do so.

IIRC the original idea was to not limit this to contrib ops (non-std ops included in ORT official releases) only; even custom ops should be allowed. I think @neginraoof wrote this part, so she can confirm.

**Comment by garymm** [comment]
@pranavsharma what is the set of ops that belongs there then?

**Comment by pranavsharma** [comment]
> @pranavsharma what is the set of ops that belongs there then?

Based on this PR https://github.com/microsoft/onnxruntime/pull/4248, it looks like this scheme was intended for contrib ops only. In that case, LGTM 👍

**Comment by pranavsharma** [comment]
You might want to change the doc here https://onnxruntime.ai/docs/tutorials/tutorials/export-pytorch-model.html. It says custom op everywhere.

File: onnxruntime/python/tools/pytorch_export_contrib_ops.py (hunk 1 above)
Comment (garymm) [reply onnxruntime/python/tools/pytorch_export_contrib_ops.py]: Switched to single underscore.

**Comment by garymm** [comment]
Can someone please approve?

> You might want to change the doc here https://onnxruntime.ai/docs/tutorials/tutorials/export-pytorch-model.html. It says custom op everywhere.

Will do. Can't do it in the same PR.

**Review by pranavsharma — DISMISSED** [review]

**Review by thiagocrepaldi — DISMISSED** [review]

**Comment by garymm** [comment]
@thiagocrepaldi could you please review again? All pipelines passed, just need approval.

File: tools/ci_build/build.py [hunk 2]
```diff
@@ -1436,6 +1436,11 @@ def run_training_python_frontend_tests(cwd):
 
     run_subprocess([
         sys.executable, '-m', 'pytest', '-sv', 'orttraining_test_orttrainer_checkpoint_functions.py'], cwd=cwd)
+    run_subprocess([
+        sys.executable, '-m', 'pytest', '-sv', 'orttraining_test_orttrainer_checkpoint_functions.py'], cwd=cwd)
```
[the remark this answers is not in the document]
Comment (garymm) [reply tools/ci_build/build.py]: Thanks for catching. Done.

**Comment by garymm** [comment]
@thiagocrepaldi could you please review again?

**Review by thiagocrepaldi — APPROVED** [review]

**Review by baijumeswani — APPROVED** [review]

**Pull request merged**